common/elasticsearch/src/main/scala/uk/ac/wellcome/elasticsearch/IndexConfig.scala

```diff
@@ -1,9 +1,33 @@
 package uk.ac.wellcome.elasticsearch
 
 import com.sksamuel.elastic4s.ElasticDsl._
-import com.sksamuel.elastic4s.requests.mappings.{FieldDefinition, ObjectField}
+import com.sksamuel.elastic4s.requests.analysis.{
+  Analysis,
+  CustomAnalyzer,
+  PathHierarchyTokenizer
+}
+import com.sksamuel.elastic4s.requests.mappings.{
+  FieldDefinition,
+  MappingDefinition,
+  ObjectField
+}
+
+trait IndexConfig {
+  val mapping: MappingDefinition
+  val analysis: Analysis
+}
+
+case object WorksIndexConfig extends IndexConfig {
+  // Analysis
+  val pathTokenizer = PathHierarchyTokenizer("path_hierarchy_tokenizer")
+  val pathAnalyzer =
+    CustomAnalyzer("path_hierarchy_analyzer", pathTokenizer.name, Nil, Nil)
+  val analysis = Analysis(
```

Comment:
I vaguely remember from the elasticsearch week that we didn't even need to use this and we could just use wildcards given the size of our index. Am I remembering correctly?

Reply:
Also, it would be nice to have this mapping change code in the PR with api code to query the collection field. This way, it's not clear what this analyzer is for

Reply:
> api code to query the collection field

Left this out specifically as we're not sure and feels like a separate place to explore - but I wanted to see the data first to explore it.

Reply:
> I vaguely remember from the elasticsearch week that we didn't even need to use this and we could just use wildcards given the size of our index. Am I remembering correctly?

We'd get _some_ query ability, and lose some.

To be clear this PR is to move us in the direction of seeing what we need, as we don't really know yet from the query perspective.